subworkflows/local/longread_preprocessing.nf

```diff
@@ -17,41 +17,32 @@ workflow LONGREAD_PREPROCESSING {
     ch_multiqc_files = Channel.empty()
 
     if ( !params.longread_qc_skipadaptertrim && params.longread_qc_skipqualityfilter) {
-        PORECHOP_PORECHOP ( reads )
-
-        ch_processed_reads = PORECHOP_PORECHOP.out.reads
-            .map { meta, reads -> [ meta + [single_end: true], reads ] }
-
-        ch_versions = ch_versions.mix(PORECHOP_PORECHOP.out.versions.first())
-        ch_multiqc_files = ch_multiqc_files.mix( PORECHOP_PORECHOP.out.log )
-
+        ch_processed_reads = LONGREAD_ADAPTERREMOVAL ( reads ).reads
+        ch_versions        = ch_versions.mix(LONGREAD_ADAPTERREMOVAL.out.versions.first())
+        ch_multiqc_files   =  ch_multiqc_files.mix( LONGREAD_ADAPTERREMOVAL.out.mqc )
     } else if ( params.longread_qc_skipadaptertrim && !params.longread_qc_skipqualityfilter) {
-
-        ch_processed_reads = FILTLONG ( reads.map { meta, reads -> [meta, [], reads ] } )
-        ch_versions = ch_versions.mix(FILTLONG.out.versions.first())
-        ch_multiqc_files = ch_multiqc_files.mix( FILTLONG.out.log )
-
+        ch_processed_reads = LONGREAD_FILTERING ( reads ).reads
+        ch_versions        = ch_versions.mix(LONGREAD_FILTERING.out.versions.first())
+        ch_multiqc_files   =  ch_multiqc_files.mix( LONGREAD_FILTERING.out.mqc )
     } else {
-        PORECHOP_PORECHOP ( reads )
-        ch_clipped_reads = PORECHOP_PORECHOP.out.reads
+        LONGREAD_ADAPTERREMOVAL ( reads )
+        ch_clipped_reads   = LONGREAD_ADAPTERREMOVAL.out.reads
             .map { meta, reads -> [ meta + [single_end: true], reads ] }
-
-        ch_processed_reads = FILTLONG ( ch_clipped_reads.map { meta, reads -> [ meta, [], reads ] } ).reads
-
-        ch_versions = ch_versions.mix(PORECHOP_PORECHOP.out.versions.first())
-        ch_versions = ch_versions.mix(FILTLONG.out.versions.first())
-        ch_multiqc_files = ch_multiqc_files.mix( PORECHOP_PORECHOP.out.log )
-        ch_multiqc_files = ch_multiqc_files.mix( FILTLONG.out.log )
+        ch_processed_reads = LONGREAD_FILTERING ( ch_clipped_reads ).reads
+        ch_versions        = ch_versions.mix(LONGREAD_ADAPTERREMOVAL.out.versions.first())
+        ch_versions        = ch_versions.mix(LONGREAD_FILTERING.out.versions.first())
+        ch_multiqc_files   =  ch_multiqc_files.mix( LONGREAD_ADAPTERREMOVAL.out.mqc )
+        ch_multiqc_files   =  ch_multiqc_files.mix( LONGREAD_FILTERING.out.mqc )
```

Comment:
Just trying to understand, is it intended behavior that if the user specifies both `longread_qc_skip_adaptertrim` and `longread_qc_skipqualityfilter` that they enter the `else` in this if block?

Reply:
Ok got it, then I'll resolve the conversation!